src/empirical.jl

```diff
@@ -45,3 +45,34 @@ function ecdf(X::RealVector{T}) where T<:Real
 end
 
 
+"""
+    epdf(X)
+
+Return an empirical probability distribution function (EPDF) based on a vector of samples
+given in `X`.
+
+Note: this is a higher-level function that returns a function, which can then be applied
+to evaluate PDF values on other samples.
+"""
+function epdf(X::AbstractVector; nbins::Int = sturges(length(X)), closed::Symbol=:left)
+    # TODO: Support edges
+    hg = normalize(fit(Histogram, X; closed=closed, nbins=nbins); mode=:pdf)
+    if hg.closed == :right
+        le = <=
+        ge = >
+    else
+        le = <
+        ge = >=
+    end
+
+    function ef(x)
+        if le(x,hg.edges[1][1]) || ge(x,hg.edges[1][end])
+            return zero(eltype(hg.weights))
+        end
+
+        idx = findfirst(le.(x,hg.edges[1]))
```

Comment:
Rather do `findfirst(y -> le(y, hg.edges[1]), x)` to avoid an allocation. Could also use `searchsortedfirst`.